mockgen/model/model.go

```diff
@@ -343,6 +343,13 @@ func typeFromType(t reflect.Type) (Type, error) {
 	}
 
 	if imp := t.PkgPath(); imp != "" {
+		// PkgPath might return a path that includes "vendor"
+		// These paths do not compile, so we need to remove everything
+		// up to and including "/vendor/"
+		// see https://github.com/golang/go/issues/12019
+		if i := strings.LastIndex(imp, "/vendor/"); i != -1 {
```

Comment:
As the referenced bug says, PkgPath is not very well-defined. If we're going to address this, we should cope with /internal/ dirs too.

I wonder if this transformation should be happening later. Or if this should be in a library somewhere else. Surely gomock isn't the only program around that wants to handle vendored import paths?